assets/mdk/proof-dependencies.mdk

```diff
@@ -0,0 +1,87 @@
+Colorizer     : dafnyx
+
+Heading Depth: 0
+
+Css Header:
+  body.madoko {
+    max-width: 100%;
+    margin: 0;
+    padding: 0;
+  }
+  .page-content {
+    padding: 0;
+  }
+
+# Introduction
+
+Problems a program's specification can undermine the trust obtained from verification. If the specification doesn't actually state what you intended it to state, the verification may still be technically sound but provide a false sense of security that the program behaves as intended. There are many ways that the text of a specification may differ from its intent, but there are some specific signs of potential mistakes that can be stated formally and reasoned about automatically. For example, if the properties you want to prove about your program are contradictory, any program will satisfy them. Similarly, If the properties you want to prove don't cover all of the program's behavior, any implementation will satisfy the uncovered portions. This latter case is frequently intentional -- you may only want to prove certain simple properties about your program, such as the lack of runtime errors -- but it can be useful to confirm that the unspecified portions do not include code you intended to specify.
```

Comment:
```suggestion
Problems in a program's specification can undermine the trust obtained from verification. If the specification doesn't actually state what you intended it to state, the verification may still be technically sound but provide a false sense of security that the program behaves as intended. There are many ways that the text of a specification may differ from its intent, but there are some specific signs of potential mistakes that can be stated formally and reasoned about automatically. For example, if the properties you want to prove about your program are contradictory, any program will satisfy them. Similarly, If the properties you want to prove don't cover all of the program's behavior, any implementation will satisfy the uncovered portions. This latter case is frequently intentional -- you may only want to prove certain simple properties about your program, such as the lack of runtime errors -- but it can be useful to confirm that the unspecified portions do not include code you intended to specify.
```

Reply:
> For example, if the properties you want to prove about your program are contradictory, any program will satisfy them

I know what you mean, but written like that it seems that if a program has contradictory ensures, any program will satisfy them, which is not true.
Could you say "contradictory requirements" instead?

Reply:
Time to the first example is a bit too long. For a blog post, I'd often have an example as soon as possible or something highlighted to think about, typically after the first paragraph. Would you have an idea on how that could be possible with your blog post?